core/src/leader_schedule_cache.rs

```diff
@@ -41,6 +50,15 @@ impl LeaderScheduleCache {
 
     pub fn slot_leader_at_else_compute(&self, slot: u64, bank: &Bank) -> Option<Pubkey> {
         let cache_result = self.slot_leader_at(slot);
+        // Forbid asking for slots in an unconfirmed epoch
```

Comment:
can you help me understand again why we have cache-aware APIs?  when do I care whether the answer comes from the cache or gets cached?

Reply:
I don't we do, I'll change it :D

Reply:
We separated into two API's to cover cases like this where .we may not have a bank available: https://github.com/solana-labs/solana/blob/master/core/src/window_service.rs#L88. Alternatively we could have just one API where the bank is an option.

Reply:
sounds better, easier to grok (for the feeble, like me)